src/NzbDrone.Core/Parser/ParsingService.cs

```diff
@@ -1,668 +1,682 @@
+        {
+            if (searchCriteria != null)
+            {
+                if (tvdbId == 0)
+                    tvdbId = _sceneMappingService.FindTvdbId(title) ?? 0;
+
+                if (tvdbId != 0 && tvdbId == searchCriteria.Series.TvdbId)
+                {
+                    return ParseSpecialEpisodeTitle(title, searchCriteria.Series);
+                }
+
+                if (tvRageId != 0 && tvRageId == searchCriteria.Series.TvRageId)
+                {
+                    return ParseSpecialEpisodeTitle(title, searchCriteria.Series);
+                }
+            }
+
+            var series = GetSeries(title);
+
+            if (series == null)
+            {
+                series = _seriesService.FindByTitleInexact(title);
+            }
+
+            if (series == null && tvdbId > 0)
+            {
+                series = _seriesService.FindByTvdbId(tvdbId);
+            }
+
+            if (series == null && tvRageId > 0)
+            {
+                series = _seriesService.FindByTvRageId(tvRageId);
+            }
+
+            if (series == null)
+            {
+                _logger.Debug("No matching series {0}", title);
+                return null;
+            }
+
+            return ParseSpecialEpisodeTitle(title, series);
+        }
+
+        private ParsedEpisodeInfo ParseSpecialEpisodeTitle(string title, Series series)
+        {
+            // find special episode in series season 0
+            var episode = _episodeService.FindEpisodeByTitle(series.Id, 0, title);
+
+            if (episode != null)
+            {
+                // create parsed info from tv episode
+                var info = new ParsedEpisodeInfo();
+                info.SeriesTitle = series.Title;
+                info.SeriesTitleInfo = new SeriesTitleInfo();
+                info.SeriesTitleInfo.Title = info.SeriesTitle;
+                info.SeasonNumber = episode.SeasonNumber;
+                info.EpisodeNumbers = new int[1] { episode.EpisodeNumber };
+                info.FullSeason = false;
+                info.Quality = QualityParser.ParseQuality(title);
+                info.ReleaseGroup = Parser.ParseReleaseGroup(title);
+                info.Language = LanguageParser.ParseLanguage(title);
+                info.Special = true;
+
+                _logger.Debug("Found special episode {0} for title '{1}'", info, title);
+                return info;
+            }
+
+            return null;
+        }
+
+        private Movie GetMovie(ParsedMovieInfo parsedMovieInfo, string imdbId, SearchCriteriaBase searchCriteria)
+        {
+            Movie movieByCriteria;
+            if ((movieByCriteria = TryFindMovieBySearchCriteria(parsedMovieInfo, searchCriteria)) != null)
+            {
+                return movieByCriteria;
+            }
+
+            Movie movieByTitle;
+            if ((movieByTitle = TryFindMovieByTitle(parsedMovieInfo, searchCriteria)) != null)

```

Comment:
Yup hate super long methods with like one local variable that is abused throughout the whole 400 lines of method scope for all kinds of nasty things. I must admit though I am using ReSharper for  this kind of breaking methods up. Just a matter of setting your cursor to the start of a large statement block and then use ReSharper 's right extend and extract methods functionality. Granted afterwards you have to do some manual adjust for return types or parameter passing with ref or out. And yeah it doesn't hurt to have a basic understanding of what is going on with the parts of the code you are looking to refactor.